Hast.Abstractions/Hast.Vitis.Abstractions/Readme.md

```diff
@@ -148,6 +148,78 @@ On the Azure virtual machine:
 3. Run the Hastlayer application with the same parameters you did on the build machine.
 
 
+## Using Hastlayer with Zynq boards
+
+This section has both generic instructions and specific steps for setting up `Hast.Samples.Consumer` for the _Trenz TE0715-04-30-1C_ module. Use the latter as template for your own projects.
+
+
+### Before you Start
+
+You need to create your own PetaLinux SD card image with the _5.4.0-xilinx-v2020.2_ kernel. You can follow [this page](https://wiki.trenz-electronic.de/display/PD/PetaLinux) on how to create the _boot.bin_, _boot.scr_ and _image.ub_ files that needs to be on your SD card. Since Hastlayer uploads the actual payload at runtime you don't need to worry about the project used. A more Hastlayer-specific setup guide is coming soon.
+
+
+### Cross Compilation
+
+You can't build _xlbin_ files on a Zynq device. So you have to cross compile on a different machine and copy the prepared files to the SD card. In case of _Hast.Samples.Consumer_ you can run it from your dev machine by calling
+```shell
+dotnet Hast.Samples.Consumer.dll \
+    -device "TE0715-04-30-1C" \
+    -sample "ParallelAlgorithm" \
+    -name "Parallel Algorithm Sample" |
+    -build
+``` 
+
+This command will get you through code transformation and build composition without actually trying to execute the result thanks to the `-build` flag. You can run it on any machine with XRT installed even if it doesn't have any FPGAs.
+
+If you are running a different project you can either introduce a flag or environment checker logic to exit after `hastlayer.GenerateHardware()` was called. Or simply accept that that application will exit with an exception on the cross compiler machine. Also you have to set the `HardwareGenerationConfiguration.SingleBinaryPath` property to ensure the executable on your device knows where to look the for _xclbin_ and its auxiliary files. You can see examples for that in _Samples/Hast.Samples.Consumer/Program.cs_.
+
+For maximum performance you should compile a Ready-to-Run build of Hastlayer. This requires a Linux host or a Linux virtual machine, for example using Docker. See more details in the _Cross Compilation with Docker_ section above.
+
+
+### SD Card Preparation Steps
+
+You need to upload some files to your microSD card to have a usable system:
+
+* Copy the _boot.bin_, _boot.scr_ and _image.ub_ files mentioned in _Before you Start_ into your SD root.
+* Download the Linux Arm32 **binary** version of the .Net 5 SDK from [here](https://dotnet.microsoft.com/download/dotnet/5.0) and extract it into a directory with the same name as the _tar.gz_ file on the SD card (e.g. _/dotnet-sdk-5.0.400-linux-arm_). This is important.
+* Copy the application directory with your Hastlayer project, for example _/Hast.Samples.Consumer_.
+* Copy all files from your _HardwareFramework/bin` directory to the card. Doesn't matter where, in this example we copy it to the _/benchmarks_ directory.

```

Comment:
Első alkalommal ez OK, de fejlesztés közben egyszerűbb lenne nem folyamatosan az SD kártyára másolgatni. Nem lehetne inkább SCP-vel csinálni ezt?

Reply:
Persze, de írtam lejjebb, hogy állítsuk be az SSH-t és az SCP/SFTP-hez más nem kell.

Reply:
Köszi. SCP-vel is gondolom az SD kártyára kell másolni, mert más storage nincs, ugye?

Reply:
Igen, gondolom a fájlrendszer többi része ramdisk. Nem vagyok ebben elég biztos ahhoz, hogy külön odaírjam de látom, hogy lekapcsolás után elveszik ami nincs a _/media/sd-mmcblk0p1_ -ben

Reply:
Igen, légyszi.